addons/mail/static/src/core/common/message.js

```diff
@@ -397,6 +397,9 @@ export class Message extends Component {
         if (ev.target.closest(".o_channel_redirect")) {
             ev.preventDefault();
             const thread = this.store.Thread.insert({ model, id });
+            if (!thread.isLocallyPinned) {
+                await this.threadService.fetchChannel(id);
+            }
```

Comment:
`isLocallyPinned = false` also suggests that the thread has `is_pinned = true` which is derived from server. Which means that the thread has been fetched from server.

So, the right condition would be to check `thread.type`, whether it is present or not.